UPGRADE-4.0.md

```diff
@@ -18,6 +18,16 @@ DependencyInjection
 
  * Using unsupported options to configure service aliases raises an exception.
 
+ * Resetting a private shared service from outside of the container with the
+   `set()` method of the `Container` class is no longer supported and will raise
+   an exception.
+
+ * Checking the existence of a private shared service with the `has()` method of
```

Comment:
Is this correct? Throwing an exception for a method that checks existence is not good for DX IMO.